pkg/network/usm/utils/connection_rollup_test.go

```diff
@@ -68,19 +69,47 @@ func TestRollupKey(t *testing.T) {
 		dstIP := util.AddressFromString("2.2.2.2")
 
 		c1 := types.NewConnectionKey(srcIP, dstIP, 6000, 80)
-		t1 := aggregator.RollupKey(c1)
+		rk1 := aggregator.RollupKey(c1)
 
 		c2 := types.NewConnectionKey(srcIP, dstIP, 6001, 80)
-		t2 := aggregator.RollupKey(c2)
+		rk2 := aggregator.RollupKey(c2)
 
 		// Let's also try a different tuple order
 		c3 := types.NewConnectionKey(dstIP, srcIP, 80, 6002)
-		t3 := aggregator.RollupKey(c3)
+		rk3 := aggregator.RollupKey(c3)
 
-		// c1, c2 and c3 should all translate to c1
-		assert.Equal(t, c1, t1)
-		assert.Equal(t, c1, t2)
-		assert.Equal(t, c1, t3)
+		c4 := types.NewConnectionKey(dstIP, srcIP, 80, 6003)
+		rk4 := aggregator.RollupKey(c4)
+
+		// Everything will be translated to essentially the same key, but tuple
+		// order will be preserved.
+		assert.Equal(t, c1, rk1)
+		assert.Equal(t, c1, rk2)
+		assert.Equal(t, c1, flipKey(rk3))
+		assert.Equal(t, c1, flipKey(rk4))
+	})
+
+	t.Run("multiple server ports", func(t *testing.T) {
```

Comment:
Could you include additional cases where we also check for the negative? For instance, when we have a different IP but the same port, and we don't perform a rollup

Reply:
Good point. Will add a test case for that! 👍